### PR TITLE
[AAE-16995] Code refactor of LocationCellComponent

### PR DESCRIPTION
**Pull request by wiktord2000**
**Please check if the PR fulfills these requirements**

> - [X] The commit message follows our [guidelines](https://github.com/Alfresco/alfresco-ng2-components/wiki/Commit-format)
> - [ ] Tests for the changes have been added (for bug fixes / features)
> - [X] Docs have been added / updated (for bug fixes / features)

**What kind of change does this PR introduce?** (check one with "x")

> - [X] Bugfix
> - [ ] Feature
> - [X] Code style update (formatting, local variables)
> - [X] Refactoring (no functional changes, no api changes)
> - [ ] Build related changes
> - [X] Documentation
> - [ ] Other... Please describe:

**What is the current behaviour?** (You can also link to an open issue here)

**What is the new behaviour?**

**Does this PR introduce a breaking change?** (check one with "x")

> - [ ] Yes
> - [X] No

If this PR contains a breaking change, please describe the impact and migration path for existing applications: ...

**Other information**:
We were trying to use this type to handle simple links but at some point we realized that it isn't the best place and we are going to compose another type in the future. But because I spent some time here I took this opportunity and made a small code refactor.

### CODE REVIEW

File: lib/core/src/lib/datatable/components/datatable-cell/datatable-cell.component.ts [hunk 1]
```diff
@@ -108,6 +93,24 @@ export class DataTableCellComponent implements OnInit, OnDestroy {
...
+                    this.row['cache'][this.column.key] = this.column.key
+                        .split('.')
+                        .reduce((source, key) => (source ? source[key] : ''), data.obj);
+
+                    this.updateValue();
+                }
+            }
+        });
+    }
+
```
Comment (tomgny) [inline lib/core/src/lib/datatable/components/datatable-cell/datatable-cell.component.ts:113]: i think we can split it more to make it more understandable, something like:

```suggestion
private subscribeToRowUpdates(): void {
    if (!this.dataTableService || !this.row.obj) {
        return;
    }

    this.dataTableService.rowUpdate.pipe(takeUntil(this.onDestroy$)).subscribe((data) => {
        if (data?.id === this.row?.id && data.obj) {
            this.row.obj = data.obj;
            this.row['cache'][this.column.key] = this.getNestedPropertyValue(data.obj, this.column.key);
            
            this.updateValue();
            }
        }
    });
}

private getNestedPropertyValue(obj: any, path: string) {
    return path.split('.').reduce((source, key) => (source ? source[key] : ''), obj);
}
```

File: lib/core/src/lib/datatable/components/datatable-cell/datatable-cell.component.ts [hunk 2]
```diff
@@ -108,6 +93,24 @@ export class DataTableCellComponent implements OnInit, OnDestroy {
         }
     }
 
+    protected subscribeToRowUpdates() {
```
Comment (tomgny) [inline lib/core/src/lib/datatable/components/datatable-cell/datatable-cell.component.ts:96]: why it's protected? shouldn't be private?

File: lib/core/src/lib/datatable/components/location-cell/location-cell.component.ts [hunk 3]
```diff
@@ -18,19 +18,24 @@
...
     template: `
         <ng-container>
-            <a href="" [title]="tooltip" [routerLink]="link">
+            <a [title]="tooltip" [routerLink]="link">
                 {{ value$ | async }}
             </a>
         </ng-container>
     `,
     encapsulation: ViewEncapsulation.None,
-    host: { class: 'adf-location-cell adf-datatable-content-cell' }
```
Comment (tomgny) [inline lib/core/src/lib/datatable/components/location-cell/location-cell.component.ts:33]: don't we need `adf-location-cell` anymore?

**Review by tomgny — COMMENTED** [review]
please check comments

File: lib/core/src/lib/datatable/components/datatable-cell/datatable-cell.component.ts (hunk 2 above)
Comment (wiktord2000) [reply lib/core/src/lib/datatable/components/datatable-cell/datatable-cell.component.ts:96]: Yes, I should be, thx

File: lib/core/src/lib/datatable/components/location-cell/location-cell.component.ts (hunk 3 above)
Comment (wiktord2000) [reply lib/core/src/lib/datatable/components/location-cell/location-cell.component.ts:33]: i've just noticed that we use it inside datatable.component.scss so I will revert it, thx

File: lib/core/src/lib/datatable/components/datatable-cell/datatable-cell.component.ts (hunk 1 above)
Comment (wiktord2000) [reply lib/core/src/lib/datatable/components/datatable-cell/datatable-cell.component.ts:113]: Good point, I'll change it, thx

**Review by tomgny — APPROVED** [review]

**Pull request merged**